File: algorithms/K/Kth Missing Positive Number/Kth Missing Positive Number.js
var findKthPositive = function(arr, k) {
    let newArr = [];
    for(let i=0,j=1; j<=arr.length+k; j++){
        arr[i]!=j?newArr.push(j):i++;
    }
    return newArr[k-1]; 
    
};

File: scripts/algorithms/K/Kth Missing Positive Number/Kth Missing Positive Number.cpp
// Runtime: 3 ms (Top 79.43%) | Memory: 9.90 MB (Top 63.69%)

class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) 
    {
        int n=arr.size();
        int low=0,high=n-1;
        while(low<=high)
        {
            int mid=(high-low)/2+low;
            int x=arr[mid]-mid-1;//will give no. of missing elements
            if(x>=k)
            high=mid-1;
            else
            low=mid+1;
        }
        return k+low;   
    }
};
